native/src/components/CalendarChoice.tsx

```diff
@@ -53,17 +48,22 @@ const CalendarChoice = ({
   const { t } = useTranslation('events')
   return (
     <Modal visible={overlayVisible} onRequestClose={closeOverlay} animationType='fade'>
-      <StyledView>
-        <TitleContainer>
-          <HeaderBackButton onPress={closeOverlay} label={t('back')} />
-          <Title>{t('chooseCalendar')}</Title>
-        </TitleContainer>
-        {calendars.map(cal => (
-          <CalendarButton key={cal.id} onPress={() => chooseCalendar(cal.id)} color={cal.color}>
-            <ButtonText>{cal.title}</ButtonText>
-          </CalendarButton>
-        ))}
-      </StyledView>
+      <Container>
+        <Header>
+          <HeaderBox goBack={closeOverlay} text={t('chooseCalendar')} textCentered />
```

Comment:
As mentioned before I think we should stay consistent with the header text and not put some user action request there when normally it always shows the title of the previous route. Therefore please move the text to a separate component directly above the calendar list.